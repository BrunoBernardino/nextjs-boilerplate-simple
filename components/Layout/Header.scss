@import '__variables';

.Header {
  display: block;
  text-align: center;
  padding: 1.3rem 0;

  @media #{$bigger-screen} {
    @include flex-spaced;
  }

  &__title {
    font-size: 3rem;
    line-height: 3rem;
    font-weight: 100;

    a {
      text-decoration: none;

      img {
        width: 120px;
      }
    }
  }

  &__nav__items {
    display: block;
    margin: 20px auto;

    @media #{$bigger-screen} {
      @include flex;
    }
  }

  &__nav__item {
    display: block;
    margin: 0 0.3rem 0.3rem;

    @media #{$bigger-screen} {
      margin: 0 0.3rem;
    }

    a {
      display: block;
      padding: 0.5rem;
      text-decoration: none;
      font-weight: bold;
      font-size: 1.1rem;
    }

    &--active {
      a {
        color: $color-pink;
      }
    }
  }
}
