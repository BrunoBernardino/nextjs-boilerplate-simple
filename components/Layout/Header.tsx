import React from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

import './Header.scss';

const Header = () => {
  const router = useRouter();
  const currentPath = router && router.route;

  return (
    <header className="Header">
      <h1 className="Header__title">
        <Link href="/">
          <a>
            <img alt="Logo: stylized letters 'BRN'" src="/images/logo.svg" />
          </a>
        </Link>
      </h1>
      <nav className="Header__nav">
        <ul className="Header__nav__items">
          <li
            className={`Header__nav__item ${
              currentPath === '/ssr' ? 'Header__nav__item--active' : ''
            }`}
          >
            <Link href="/ssr">
              <a>SSR</a>
            </Link>
          </li>
          <li
            className={`Header__nav__item ${
              currentPath === '/sg' ? 'Header__nav__item--active' : ''
            }`}
          >
            <Link href="/sg">
              <a>SG</a>
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
