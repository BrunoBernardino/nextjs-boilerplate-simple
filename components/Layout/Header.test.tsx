import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import Header from './Header';

expect.extend(enzymify());

describe('Header', () => {
  it('renders the Header without errors', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('nav').length).toEqual(1);
    expect(wrapper.find('Link[href="/ssr"]').length).toEqual(1);
    expect(wrapper.find('Link[href="/sg"]').length).toEqual(1);
  });

  it('renders the Header as expected', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
