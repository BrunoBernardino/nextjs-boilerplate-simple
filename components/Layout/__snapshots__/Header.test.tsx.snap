// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header as expected 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__title"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        alt="Logo: stylized letters 'BRN'"
        src="/images/logo.svg"
      />
    </a>
  </h1>
  <nav
    className="Header__nav"
  >
    <ul
      className="Header__nav__items"
    >
      <li
        className="Header__nav__item "
      >
        <a
          href="/ssr"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          SSR
        </a>
      </li>
      <li
        className="Header__nav__item "
      >
        <a
          href="/sg"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          SG
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
